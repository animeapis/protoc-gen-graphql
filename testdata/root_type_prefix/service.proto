syntax = "proto3";

package protoc_gen_graphql.test.grpc;

import "graphql/options.proto";

option go_package = "github.com/animeapis/protoc-gen-graphql/testdata;testdata";

service Service {
  option (graphql.service) = { reference_name: "service_name" };

  rpc ReadMethod(Request) returns (Response) {
    option (graphql.method) = { operation: "query" };
  }

  rpc WriteMethod(Request) returns (Response) {
    option (graphql.method) = { operation: "mutation" };
  }
}

message Request {
  string request_name = 1;
}

message Response {
  string response_name = 1;
}
