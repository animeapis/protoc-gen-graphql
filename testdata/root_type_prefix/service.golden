# DO NOT EDIT! Generated by protoc-gen-graphql.

extend type Query {
  serviceName: ProtocGenGraphqlTestGrpc_Service_Query!
}

type ProtocGenGraphqlTestGrpc_Service_Query {
  readMethod(input: ProtocGenGraphqlTestGrpc_RequestInput!): ProtocGenGraphqlTestGrpc_Response!
}

extend type Mutation {
  serviceName: ProtocGenGraphqlTestGrpc_Service_Mutation!
}

type ProtocGenGraphqlTestGrpc_Service_Mutation {
  writeMethod(input: ProtocGenGraphqlTestGrpc_RequestInput!): ProtocGenGraphqlTestGrpc_Response!
}

type ProtocGenGraphqlTestGrpc_Request {
  requestName: String!
}

input ProtocGenGraphqlTestGrpc_RequestInput {
  requestName: String
}

type ProtocGenGraphqlTestGrpc_Response {
  responseName: String!
}
